crepecake/src/main/java/com/google/cloud/tools/crepecake/registry/RegistryClient.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.registry;
+
+import com.google.api.client.http.HttpResponseException;
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.http.Authorization;
+import com.google.cloud.tools.crepecake.http.Connection;
+import com.google.cloud.tools.crepecake.http.Request;
+import com.google.cloud.tools.crepecake.http.Response;
+import com.google.cloud.tools.crepecake.image.DescriptorDigest;
+import com.google.cloud.tools.crepecake.image.json.ManifestTemplate;
+import com.google.cloud.tools.crepecake.json.JsonTemplateMapper;
+import com.google.cloud.tools.crepecake.registry.json.ErrorEntryTemplate;
+import com.google.cloud.tools.crepecake.registry.json.ErrorResponseTemplate;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import javax.annotation.Nullable;
+
+/** Interfaces with a registry. */
+public class RegistryClient {
+
+  // TODO: This should be configurable.
+  private static final String PROTOCOL = "http";
```

Comment:
why not https?

Reply:
This will be taken care of in https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/78. Using HTTPS directly here doesn't work with local registries.